tmt/steps/provision/__init__.py

```diff
@@ -464,25 +463,32 @@ def stop(self):
 
         raise NotImplementedError()
 
-    def reboot(self, hard=False):
+    def reboot(self, hard=False, command=None, timeout=None):
         """
         Reboot the guest, return True if successful
 
         Parameter 'hard' set to True means that guest should be
         rebooted by way which is not clean in sense that data can be
         lost. When set to False reboot should be done gracefully.
+
+        Use the 'command' parameter to specify a custom reboot command
+        instead of the default 'reboot'.
+
+        Parameter 'timeout' can be used to specify time (in seconds) to
+        wait for the guest to come back up after rebooting.
         """
 
         raise NotImplementedError()
 
-    def reconnect(self, timeout=CONNECTION_TIMEOUT):
+    def reconnect(self, timeout=None):
         """
         Ensure the connection to the guest is working after reboot
 
         The default timeout is 1h to allow long operations such as
         system upgrade. Custom number of seconds can be provided in the
         `timeout` parameter.
         """
+        timeout = timeout or CONNECTION_TIMEOUT
```

Comment:
I'd like to explain my thought process for this change in advance before reviewers come.

IMHO, the execute plugin should not have to worry about the default timeout values. If an empty value comes in the reboot request file, it should just pass `None` to `reboot()`. With the default argument value being `CONNECTION_TIMEOUT`, the `reboot` method would have to do something like:

```
if timeout:
    self.reconnect(timeout)
else:
    self.reconnect()
```

which is not great, the default should be handled in one place. That's why I changed the default value to `None`. This way, the handling is simple and at the same time can be overridden (see `GuestContainer`).

Reply:
Will do. Just wanted to see if there is a better way to accomplish this, I currently can't think of anything much better :)

Reply:
The default value is hidden in the body of the function, which is a bit weird. I'm not sure I fully understand the problem you wanted to solve, I have a strange feeling it's slipping through my fingers :) If an empty value comes in the reboot request file, you do not want this `None` to override the default value of `timeout=CONNECTION_TIMEOUT` when calling `guest.reboot(timeout=...)` in `_handle_reboot()`, is that correct?

Reply:
Yes, you summarized the problem better than me :)